gulpfile.js

```diff
@@ -89,6 +89,10 @@
 
     gulp.task(
         'package:clean',
+        [
+            'qa',
```

Comment:
tl;dr This makes the `gulp package` task atomic.

In a way it's not dependent, but if you think about the commands actually used, and the way `gulp` processes, I think it actually makes more sense to have it in here.

The user isn't likely (nor instructed) to run `gulp package:clean`  directly, although I agree that if they do, they will be prevented from removing a directory because of a failing test.  However they can just do `rm -rf dist/[env] ; mkdir -p dist/[env]`, that's all the task does anyway.

The more likely (and intended, documented) usage is to run `gulp package`.  In that case, `package` will fire off its dependencies, including `qa`, `build` and `package:clean`.  If `package:clean` also has these dependencies, then it will also wait for `qa` and `build` to complete successfully, and therefore the old `dist/[env]` isn't deleted until the `qa` and `build` tasks succeed (and not if they fail).  If these dependencies aren't here, then `package:clean` will just go ahead regardless of `qa` and `build`, so if the tests fail, you'll end up with an empty `dist/[env]`, which isn't what you want. This way, you either get the new package or keep the old one, when you run `gulp package` (and you know which one you have via the output and exit status).